src/main/java/task/Todo.java

```diff
@@ -0,0 +1,105 @@
+package task;
+
+import task.Task;
+
+/**
+ * Represents a todo under Task class
+ * Allows users to get_type and get_task
+ */
+public class Todo extends Task {
+    private String type;
+    private String dateTime;
+    private String task;
+
+
+
+    /**
+     * Constructor to create a todo task for newly inputed messages
+     * Sieve out the task and date separately
+     *
+     * @param message String the user input message
+     */
+    public Todo (
+            String message) {
+        super(message);
+        this.setDateTime();
+        this.setTask();
+        this.setType();
+    }
+
+
+    /**
+     * Constructor to create a todo task for Duke.txt lines
+     * Sieve out the task and date separately
+     *
+     * @param message String the lines in Duke.txt file
+     * @param isDuke Boolean to differentiate between 2 constructors, always true
+     */
+    public Todo (
+            String message,
+                Boolean isDuke) {
+        super(message);
+        this.setDateTime();
+        this.setTask2();
+        this.setType();
+    }
+
+    @Override
+    public void setType() {
+        this.type = "T";
+    }
+
+    /**
+     * Sieve out the task + date portion of the message
+     * Only for newly inputed messages
+     *
+     */
+    @Override
+    public void setTask() {
```

Comment:
Maybe you can categorise both the different the setTask methods into one by specifying arguments